explorer/deploy/webAppCreator.ts

```diff
@@ -543,4 +569,26 @@ class WebsiteStep extends WebAppCreatorStepBase {
         }
     }
 
+    //Implements new Service principal model for ACR container registries while maintaining old admin enabled use
+    private async acquireRegistryLoginCredentials(): Promise<void> {
+        if (this._serverPassword && this._serverUserName) { return; }
+
+        const client = new ContainerRegistryManagementClient(this.azureAccount.getCredentialByTenantId(this._imageSubscription.tenantId), this._imageSubscription.subscriptionId);
+        const resourceGroup: string = this._registry.id.slice(this._registry.id.search('resourceGroups/') + 'resourceGroups/'.length, this._registry.id.search('/providers/'));
+
+        if (this._registry.adminUserEnabled) {
+            try {
+                let creds = await client.registries.listCredentials(resourceGroup, this._registry.name);
+                this._serverPassword = creds.passwords[0].value;
+                this._serverUserName = creds.username;
+                return;
+            } catch (error) {
+                vscode.window.showErrorMessage(error.message);
```

Comment:
await?

Reply:
In this case await pauses execution until the user clicks to exit the window which is not a guarantee, additionally it makes no difference as the error is shown when ready and execution ends nonetheless.

Reply:
We shouldn't need to both show a message and throw an error - the error should get displayed by the caller, is that not happening?

One issue is that this code was copied from elsewhere and we need to actually update it to use the shared code in https://github.com/Microsoft/vscode-azuretools, so maybe that needs to be done before integrating these changes.

Reply:
This does not happen indeed, in fact there is no visible information about an error occurring if I propagate it all the way to the code I didn't write, although I could modify it to act in this way. Additionally, while I agree that we should update the original code for the whole deploy to app services, this PR is essentially an important bugfix which by the nature of the bug broke the existing app services functionality and thus required me to solve a small number of issues on that side to keep existing functionality post the PR. Would it not be best to integrate this functional code with the added bugfixes before attempting the necessary refactoring that appears to me as less pressing?

Reply:
Makes sense.  Added https://github.com/Microsoft/vscode-docker/issues/358 to track that we want telemetry for such failures.

BTW, with my merge of https://github.com/Microsoft/vscode-docker/pull/350, the caller should now be catching and handling (telemetry and error messages) anything you throw from commands.  That should now be the correct way to handle things (let exceptions bubble up), except for cases like there where you need to continue.

Reply:
sweet, Ill update accordingly